### PR TITLE
crd watchfilter patch via kustomize

### PR DESCRIPTION
**Pull request by bavarianbidi**
drop 50 Lines of bash-code to work around the missing `wildcard` support in kustomize and replace it with 20 lines of kustomize configuration.

Open tasks:
- [ ] add version check for `kustomize >= v5.0`
- [ ] check CI
- [ ] rename `generate-kustomize-patches.sh`
- [ ] CHANGELOG

### Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: Makefile.custom.mk [hunk 1]
```diff
@@ -3,6 +3,7 @@ generate: ## Generate kustomize patches and all helm charts
 	./hack/generate-kustomize-patches.sh
 	$(MAKE) delete-generated-helm-charts
 	kustomize build config/helm -o helm/cluster-api/templates
+	rm -v helm/cluster-api/templates/v1_configmap_watchfilter-patch.yaml
```
Comment (njuettner) [inline Makefile.custom.mk:6]: According to the checks `helm/watchfilter-patch-configmap.yaml` does not exist. Is that right?

Comment (bavarianbidi) [reply Makefile.custom.mk:6]: thanks for the hint - fixed

**Review by njuettner — APPROVED** [review]
lgtm

**Comment by njuettner** [comment]
Just one note, don't forget to update the CHANGELOG 🙂

**Pull request merged**